refarch-gateway/pom.xml

```
@@ -18,7 +18,9 @@
     <name>refarch-gateway</name>
 
     <properties>
-        <java.version>17</java.version>
+        <java.version>21</java.version>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
```

To keep this in sync with the other Java based projects I suggest using `java.version` as a variable and additionally set `maven.compiler.release`

```suggestion
        <maven.compiler.source>${java.version}</maven.compiler.source>
        <maven.compiler.target>${java.version}</maven.compiler.target>
        <maven.compiler.release>${java.version}</maven.compiler.release>
```